WooCommerce/WooCommerceTests/Tools/InfiniteScroll/PaginationTrackerTests.swift

```diff
@@ -0,0 +1,120 @@
+import XCTest
+@testable import WooCommerce
+
+final class PaginationTrackerTests: XCTestCase {
+    private let paginationTracker = PaginationTracker(pageFirstIndex: 0, pageSize: 30)
+    private let mockDelegate = MockPaginationTrackerDelegate()
+
+    override func setUp() {
+        super.setUp()
+        paginationTracker.delegate = mockDelegate
+    }
```

Comment:
⚠️  We should do a `tearDown` even though these objects are light. 🙂 

```suggestion
    private var paginationTracker: PaginationTracker!
    private var mockDelegate: MockPaginationTrackerDelegate!

    override func setUp() {
        super.setUp()
        mockDelegate = MockPaginationTrackerDelegate()
        paginationTracker = PaginationTracker(pageFirstIndex: 0, pageSize: 30)
        paginationTracker.delegate = mockDelegate
    }

    override func tearDown() {
        paginationTracker = nil
        mockDelegate = nil
        super.tearDown()
    }
```

Reply:
sure sounds good I'll keep this in mind next time! updated in https://github.com/woocommerce/woocommerce-ios/pull/2750/commits/d3a058fd52f17405824e28fedfb66a8665eee178